account_ebics_batch/models/ebics_batch_log.py

```diff
@@ -0,0 +1,186 @@
+                ("state", "=", "active"),
+            ]
+        )
+        log = log_model.create(
+            {
+                "ebics_config_ids": [(6, 0, configs.ids)],
+                "date_from": date_from,
+                "date_to": date_to,
+            }
+        )
+        ebics_file_ids = []
+        for config in configs:
+            err_msg = (
+                _("Error while processing EBICS connection '%s' :\n") % config.name
+            )
+            if config.state == "draft":
+                import_dict["errors"].append(
+                    err_msg
+                    + _(
+                        "Please set state to 'Confirm' and "
+                        "Reprocess this EBICS Import Log."
+                    )
+                )
+                continue
+            try:
+                with self.env.cr.savepoint():
+                    ebics_file_ids += self._ebics_import(
+                        config, date_from, date_to, import_dict
+                    )
+            except UserError as e:
+                import_dict["errors"].append(err_msg + " ".join(e.args))
+            except Exception:
+                tb = "".join(format_exception(*exc_info()))
+                import_dict["errors"].append(err_msg + tb)
+        log.file_ids = [(6, 0, ebics_file_ids)]
+        try:
+            with self.env.cr.savepoint():
+                log._ebics_process(import_dict)
+        except UserError as e:
+            import_dict["errors"].append(err_msg + " ".join(e.args))
+        except Exception:
+            tb = "".join(format_exception(*exc_info()))
+            import_dict["errors"].append(err_msg + tb)
+        log._finalise_processing(import_dict)
+
+    def _finalise_processing(self, import_dict):
+        log_item_model = self.env["ebics.batch.log.item"]
+        state = self.has_draft_files and "draft" or "done"
+        note = ""
+        error_count = 0
+        if import_dict["errors"]:
+            state = "error"
+            note = "\n\n".join(import_dict["errors"])
+            error_count = len(import_dict["errors"])
+        log_item_model.create(
+            {
+                "log_id": self.id,
+                "state": state,
+                "note": note,
+                "error_count": error_count,
+            }
+        )
+        self.state = state
+
+    def _ebics_import(self, config, date_from, date_to, import_dict):
+        ctx = dict(self.env.context, ebics_download=True)
+        xfer_wiz = (
+            self.env["ebics.xfer"]
+            .with_context(**ctx)
+            .create(
+                {
+                    "ebics_config_id": config.id,
+                    "date_from": date_from,
+                    "date_to": date_to,
+                }
+            )
+        )
+        xfer_wiz._onchange_ebics_config_id()
+        res = xfer_wiz.ebics_download()
+        file_ids = res["context"].get("ebics_file_ids")
+        if res["context"]["err_cnt"]:
```

Comment:
`err_cnt` isn't existing. If it's set sometimes it's better to use `res["context"].get("err_cnt")` otherwise there must be another way to achieve this.

Reply:
it's set in https://github.com/Noviat/account_ebics/blob/4a10e40b33e5a403b2f6f2df2b36f17e8a812a1b/account_ebics/wizards/ebics_xfer.py#L180
You'll have to backport also that part, otherwise I don't see how you can get a reliable cron operation.